### PR TITLE
Fetch latest HTX build automatically for Host and peer

### PR DESCRIPTION
**Pull request by FarooqAbdulla01**
This code supports to fetch the latest available HTX builds on RHEL for both Host and Peer as per RHEL OS version.
Ex: 1. For RHEL-8 installs RHEL-8 based HTX rpm.
    2. For RHEL-9 installs RHEL-9 based HTX rpm.

### CODE REVIEW

**Comment by FarooqAbdulla01** [comment]
Tested with 
1. Host LPAR as RHEL 8 and Peer LPAR as RHEL 9
2. Host LPAR as RHEL 8 and Peer LPAR as RHEL 8 
Both conditions version based installation getting sucessful.
Currently this code does not support for SLES remote installations and separate JIRA has been initiated on same.
[htx_nic_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/11329614/htx_nic_job.log)
 

[root@ltcden4-lp4 net]# avocado run htx_nic_devices.py -m htx_nic_devices.py.data/htx_nic_devices.yaml --max-parallel-tasks=1
JOB ID     : af1bb74be8b9a6b58acc6d5fdb780534e4296d56
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2023-04-25T09.00-af1bb74/job.log
Task exception was never retrieved
future: <Task finished coro=<StatusServer.serve_forever() done, defined at /usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/status/server.py:40> exception=AttributeError("'Server' object has no attribute 'serve_forever'",)>
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/status/server.py", line 44, in serve_forever
    await self._server_task.serve_forever()
AttributeError: 'Server' object has no attribute 'serve_forever'
 (1/3) htx_nic_devices.py:HtxNicTest.test_start;run-0e2b: STARTED
 (1/3) htx_nic_devices.py:HtxNicTest.test_start;run-0e2b: PASS (637.59 s)
 (2/3) htx_nic_devices.py:HtxNicTest.test_check;run-0e2b: STARTED
 (2/3) htx_nic_devices.py:HtxNicTest.test_check;run-0e2b: PASS (123.42 s)
 (3/3) htx_nic_devices.py:HtxNicTest.test_stop;run-0e2b: STARTED
 (3/3) htx_nic_devices.py:HtxNicTest.test_stop;run-0e2b: PASS (11.43 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2023-04-25T09.00-af1bb74/results.html

**Review by PraveenPenguin — APPROVED** [review]

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Comment by Naresh-ibm** [comment]
@FarooqAbdulla01 what about the sles disro versions?
will it work for sles too or we need to handle it differently again?

File: io/net/htx_nic_devices.py [hunk 1]
```diff
@@ -95,13 +98,25 @@ def get_peer_distro(self):
...
         """
         packages = ['git', 'gcc', 'make', 'wget']
         detected_distro = distro.detect()
-        if detected_distro.name in ['centos', 'fedora', 'rhel', 'redhat', 'wget']:
+        if detected_distro.name in [
+                                    'centos',
+                                    'fedora',
+                                    'rhel',
+                                    'redhat',
+                                    'wget']:
```
[the remark this answers is not in the document]
Comment (FarooqAbdulla01) [reply io/net/htx_nic_devices.py]: @Naresh-ibm removed as its not required

File: io/net/htx_nic_devices.py.data/htx_nic_devices.yaml [hunk 2]
```diff
@@ -11,4 +11,4 @@ peer_ips: ""
 net_ids: ""
 # time limit in minutes
 time_limit: 2
-htx_rpm: ""
+rpm_link: ""
```
[the remark this answers is not in the document]
Comment (FarooqAbdulla01) [reply io/net/htx_nic_devices.py.data/htx_nic_devices.yaml]: @Naresh-ibm changed htx_rpm to htx_rpm_link and embedded "htx rpm" URL location in yaml file.

File: io/net/htx_nic_devices.py [hunk 3]
```diff
@@ -122,13 +137,43 @@ def build_htx(self):
...
+                        self.cancel("Installion of rpm failed")
+
+                if pattern == peer_distro_pattern:
+                    cmd = ('rpm -ivh --nodeps %s%s '
+                           '--force' % (self.rpm_link, self.latest_htx_rpm))
+                    output = self.session.cmd(cmd)
+                    if not output.exit_status == 0:
+                        self.cancel("Unable to install the package %s %s"
+                                    " on peer machine" % (self.rpm_link,
+                                                          self.latest_htx_rpm))
```
[the remark this answers is not in the document]
Comment (FarooqAbdulla01) [reply io/net/htx_nic_devices.py:172]: @Naresh-ibm yes.we are installing the RPM package on Peer machine according to the rhel version.
please check the logs attached.

**Comment by FarooqAbdulla01** [comment]
> @Naresh-ibm 
basically there are 2 problems when dealing with SLES.
1. when SLES/Host and SLES/peer ---> Getting "password" prompt issue, will be opening separate  JIRA for tracking.
2. when RH/Host and SLES/Peer  ------> Getting installation type issue, like when trying to install the package on remote peer 
instead of selecting "zypper" to install package the script was selecting "dnf" - This needed to be corrected - Opening separate JIRA for tracking.

File: io/net/htx_nic_devices.py [hunk 4]
```diff
@@ -408,7 +454,8 @@ def htx_configure_net(self):
                 else:
                     self.session.cmd("systemctl restart network")
                 if self.host_distro == "rhel":
-                    process.system("systemctl start NetworkManager", shell=True,
+                    process.system("systemctl start NetworkManager",
+                                   shell=True,
```
[the remark this answers is not in the document]
Comment (FarooqAbdulla01) [reply io/net/htx_nic_devices.py:454]: @Naresh-ibm corrected according to pycodestyle.

**Comment by abdhaleegit** [comment]
@Naresh-ibm do you have any more comments Please ACK

File: io/net/htx_nic_devices.py.data/htx_nic_devices.yaml [hunk 5]
```diff
@@ -11,4 +11,4 @@ peer_ips: ""
 net_ids: ""
 # time limit in minutes
 time_limit: 2
-htx_rpm: ""
+htx_rpm_link: "http://ausgsa.ibm.com//gsa/ausgsa/projects/h/htx/public_html/htxonly/"
```
Comment (abdhaleegit) [inline io/net/htx_nic_devices.py.data/htx_nic_devices.yaml:14]: Do not change the parameter name keep it htx_rpm  the reason
1. htx block device uses htx_rpm as param
2. all my CR input files has htx_rpm as input, I cannot change them all.. also  if a new param gets added ..than it is duplicate one extra param

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
keep same name htx_rpm

File: io/net/htx_nic_devices.py.data/htx_nic_devices.yaml (hunk 5 above)
Comment (FarooqAbdulla01) [reply io/net/htx_nic_devices.py.data/htx_nic_devices.yaml:14]: @abdhaleegit sure, will maintain same param.

File: io/net/htx_nic_devices.py [hunk 6]
```diff
@@ -71,20 +71,23 @@ def setUp(self):
                     networkinterface.save(ipaddr, self.netmask)
                 networkinterface.bring_up()
         self.host_distro = distro.detect()
```
Comment (nasastry) [inline io/net/htx_nic_devices.py:73]: There is no need of self here for `self.host_distro` this variable is not used later.

**Comment by FarooqAbdulla01** [comment]
Getting issues while pushing the latest code this PR, hence closing this PR and opened "Fetch latest HTX build automatically for Host and peer #2667" for the same with latest code data.

**Pull request closed without merging**